script.js

```diff
@@ -1,4 +1,216 @@
+
+  userRPS = convertToRPS(usrPickNum);
+  //get randomized pick
+
+  var compPickNum = computerRPS();
+  compRPS = convertToRPS(compPickNum);
+  console.log("computer pick", compPickNum, compRPS);
+
+  //compare result
+
+  if (gameMode_Current == gameMode_Reverse)
+    return `Game Mode: Reverse Rock-Paper-Scissors. <br><br> ${rCompareRPSResult(
+      usrPickNum,
+      compPickNum
+    )}`;
+  return `Game Mode: Reverse Rock-Paper-Scissors. <br><br> ${compareRPSResult(
+    usrPickNum,
+    compPickNum
+  )}`;
+};
+
+//----MAIN
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (input == null || input == "") return "Please enter a name.";
+
+  //gets users name
+  if (gameMode_Current == gameMode_Name) {
+    userName = input;
+    gameMode_Current = gameMode_GameSel; //change game mode to game select
+    return `Welcome ${userName}. <br><br>
+    Please select a gamemode: <br>
+    -type 'rps' for Rock-Paper-Scissors<br>
+    -type 'rev' for reverse Rock-Paper-Scissors`;
+  } else {
+    if (gameMode_Current == gameMode_GameSel) {
+      if (input.toLowerCase() == "rps") {
+        gameMode_Current = gameMode_RPS;
+        return "Game Mode: Rock-Paper-Scissors<br> Type r|rock, p|paper, s|scissors";
+      } else if (input.toLowerCase() == "rev") {
+        gameMode_Current = gameMode_Reverse;
+        return "Game Mode: Reverse Rock-Paper-Scissors<br> Type r|rock, p|paper, s|scissors";
+      } else
+        return `Hi ${userName}, please enter a game mode. <br> 
+      -type 'rps' for Rock-Paper-Scissors<br>
+      -type 'rev' for reverse Rock-Paper-Scissors`;
+    }
+    //game proper
+    else {
+      //game mode: change name
+      if (input.toLowerCase() == "name") {
+        gameMode_Current = gameMode_Name;
+        return `Game Mode: Naming<br>
+        Please Enter your name.`;
+      }
+      //game mode: change mode to rps
+      else if (input.toLowerCase() == "rps") {
+        if (gameMode_Current == gameMode_RPS)
+          return `Game Mode is already Rock-Paper-Scissors.<br>
+        Type r|rock, p|paper, s|scissors`;
+
+        gameMode_Current == gameMode_RPS;
+        return `Game Mode: Game Select<br>
+        Please Enter your game. <br>
+        -type 'rps' for Rock-Paper-Scissors<br>
+        -type 'rev' for reverse Rock-Paper-Scissors`;
+      }
+      //game mode: change mode to reverse
+      else if (input.toLowerCase() == "rev") {
+        if (gameMode_Current == gameMode_Reverse)
+          return `Game Mode is already Reverse Rock-Paper-Scissors.<br>
+        Type r|rock, p|paper, s|scissors`;
+
+        gameMode_Current == gameMode_Reverse;
+        return `Game Mode: Game Select<br>
+        Please Enter your game. <br>
+        -type 'rps' for Rock-Paper-Scissors<br>
+        -type 'rev' for reverse Rock-Paper-Scissors`;
+      }
+      //continue game
+      else return rpsGame(input);
+    }
```

Comment:
You could also attempt the more advanced `switch` statement for a cleaner game mode selection in the main function:
https://www.w3schools.com/js/js_switch.asp